###  Macro to Simplify Array Length Calculation (Revived from PR #401)

arturobernalg opened the pull request:
This PR is based on [PR #401](https://github.com/apache/httpd/pull/401), which proposed 
adding the `ARRAY_LEN` macro to improve code readability and maintainability. The original 
PR was abandoned and had conflicts, so I have rebased it, resolved issues, and integrated 
the changes.

arturobernalg committed:
Replace manual array size calculation with AP_ARRAY_LEN macro

This commit introduces the AP_ARRAY_LEN macro to simplify array length calculations
by replacing occurrences of `sizeof(a)/sizeof(a[0])` with `AP_ARRAY_LEN(a)`.

notroj commented:
This looks better - thanks for redoing it - but it should be AP_ARRAY_LEN() rather than just ARRAY_LEN() to avoid namespace conflicts. Hopefully you can change it with a sed command :)

arturobernalg commented:
> This looks better - thanks for redoing it - but it should be AP_ARRAY_LEN() rather than just ARRAY_LEN() to avoid namespace conflicts. Hopefully you can change it with a sed command :)



> This looks better - thanks for redoing it - but it should be AP_ARRAY_LEN() rather than just ARRAY_LEN() to avoid namespace conflicts. Hopefully you can change it with a sed command :)

Agreed. I've implemented the changes and updated the committee message.

thomasmey commented:
Also thanks from my side!

notroj commented:
Can you make your fork writable by committers? I'd like to test a couple of additional changes to this.
